src/event_system/events.ts

```diff
@@ -0,0 +1,105 @@
+import Queue = require('better-queue')
+import { StoreOptions } from "better-queue"
+import { ConnectionStringParser } from "connection-string-parser";
+import { EventT, EventsT } from "../types/events/eventT";
+import { RegisteredEventT, RegisteredEventsT } from "../types/events/registered_eventT";
+import { TaskT, TasksT } from "../types/events/taskT";
+import EventStorage from "../data_storage/events/event_storage";
+import QueueStorage from "../data_storage/events/queue_storage";
+import Result from "../result";
+import Logger from "../logger";
+import axios, { AxiosResponse } from "axios";
+import Config from "../config";
+
+export class QueueProcessor {
+
+  private static instance: QueueProcessor;
+
+  public static get Instance(): QueueProcessor {
+    if (!QueueProcessor.instance) {
+      QueueProcessor.instance = new QueueProcessor()
+    }
+
+    return QueueProcessor.instance
+  }
+
+  private store = makeStore();
+
+  // define the database queue
+  private messageQueue: Queue = new Queue(() => undefined, {
+    store: this.store
+  });
+
+  public addEvents(events: EventsT) {
+    if (Config.queue_system === 'database') {
+      this.messageQueue.push(events)
+    } else {
+      Logger.error(`queue system ${Config.queue_system} unrecognized. Please supply supported value.`)
+    }
+  }
+}
+
+export async function StartQueue(): Promise<Result<boolean>> {
+  Logger.debug('starting queue listener for event system');
+
+  // process the tasks stored in the database queue. Each task may have multiple events
+  if (Config.queue_system === 'database') {
+    while (true) {
+      const tasks: TaskT[] = await QueueStorage.Instance.List();
+      for (const task of tasks) {
+        const events: EventT[] = task.task;
+        for (const event of events) {
+          let registeredEvents: RegisteredEventsT;
+          if (event.source_type === 'data source') {
+            registeredEvents = (await EventStorage.Instance.ListByDataSource(event.type, event.source_id)).value
```

Comment:
generally check the response value for the `isError` value. If you attempt to access the `value` property and the process has errored it will throw an uncaught exception.

Reply:
What do you think should be the behavior here for an error? Log the error or take more drastic action?

Reply:
Just log